qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -192,7 +194,70 @@ public void test52BundledJdkRemoved() throws IOException {
         }
     }
 
-    public void test60AutoCreateKeystore() {
+    public void test53JavaHomeWithSpecialCharacters() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+
+        Platforms.onWindows(() -> {
+            final Shell sh = new Shell();
+            final String originalPath = sh.run("$Env:PATH").stdout.trim();
+            final String javaHome = sh.run("$Env:JAVA_HOME").stdout.trim();
+
+            try {
+                // once windows 2012 is no longer supported and powershell 5.0 is always available we can change this command
+                sh.runIgnoreExitCode("cmd /c mklink /D 'C:\\Program Files (x86)\\java' $Env:JAVA_HOME");
+
+                sh.getEnv().put("JAVA_HOME", "C:\\Program Files (x86)\\java");
+
+                //verify ES can start, stop and run plugin list
+                Archives.runElasticsearch(installation, sh);
+
+                Archives.stopElasticsearch(installation);
+
+                String pluginListCommand = installation.bin + "/elasticsearch-plugin list";
+                Result result = sh.run(pluginListCommand);
+                assertThat(result.exitCode, equalTo(0));
+
+            } finally {
+                //clean up sym link
+                sh.runIgnoreExitCode("cmd /c del /F /Q 'C:\\Program Files (x86)\\java' ");
+                sh.getEnv().put("JAVA_HOME", javaHome);
+            }
+        });
+
+        Platforms.onLinux(() -> {
+            final Shell sh = new Shell();
+            final String javaHome = sh.run("echo $JAVA_HOME").stdout.trim();
+
+            try {
+
+                // Create temporary directory with a space and link to java binary.
+                // Use it as java_home
+                final String temp = sh.runIgnoreExitCode("mktemp -d --suffix=\"java home\"").stdout.trim();
```

Comment:
Create a temp directory seems error prone, since it may have noexec on some systems. I think this test can use any arbitrary path since it will clean it up afterwards?

Reply:
we create temp directories when starting up as well. If some systems are not supporting this, it would be better to catch that during testing.
Do you have any specific systems we support in mind that would fail on this?

Reply:
Yes we do create temp directories, but we don't try executing anything from them. Why we do we need this to be complicated creating a special temp dir when we can just use an arbitrary path and clean it up afterwards?

Reply:
I see, we can use a regular directory for this. Should be easy. I was thinking that `mktemp` is ok as we use it through a lot of `bats` tests at the moment. Will keep an eye on further migration to avoid it.